src/averageAges.js

```diff
@@ -59,6 +91,65 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+
+  // 'Average age difference with son'
+
+  const mothers = people
+    .filter((person) => {
+      return !onlyWithSon
+        ? person.sex === 'f'
+        : person.sex === 'f'
+        && people.some((p) => p.mother === person.name
+          && p.sex === 'm');
+    });
+
+  const children = people
+    .filter((person) => {
+      return !onlyWithSon
+        ? person.mother !== null
+        : person.sex === 'm'
+        && people
+          .some((p) => person.mother === p.name);
+    });
+
+  const yearsOldMom = children.map(borned => {
+    const momsAges = mothers
+      .map((year) => borned.born - year.born);
+    const averageMomsAge = momsAges
+      .reduce((sum, age) => sum + age, 0) / momsAges.length;
+
+    return averageMomsAge;
+  });
+
+  const sumAgeMother = yearsOldMom
+    .reduce((prev, year) => prev + year, 0);
+
+  const ageChildren = children
+    .map((year) => year.died - year.born)
+    .reduce((prev, age) => prev + age, 0);
+
+  const resultDiffSon
+    = (ageChildren / children.length) - (sumAgeMother / mothers.length);
+
+  const ageDiff = children
+    .map((child) => {
+      const mother = mothers
+        .find((person) => person.name === child.mother);
+
+      return mother
+        ? child.born - mother.born
+        : 0;
+    });
+
+  const totalAgeDiff = ageDiff.reduce((total, age) => total + age, 0);
+
+  const averageAgeDiff = totalAgeDiff / mothers.length;
+
+  const resultDif = onlyWithSon
+    ? resultDiffSon
+    : averageAgeDiff;
+
+  return resultDif;
```

Comment:
i'm sure it can be simplified to 
get children => write child mothers to elements => calculate average ages